app/models/miq_server/monitor/kill.rb

```diff
@@ -0,0 +1,40 @@
+module MiqServer::WorkerManagement::Base::Monitor::Kill
+  extend ActiveSupport::Concern
+
+  def kill_all_workers
+    return unless my_server.is_local?
+
+    remove_unknown_workers
+    remove_workers
+  end
+
+  private
+
+  def remove_unknown_workers
+    # We try, but forget to write migrations to delete orphaned worker class rows, causing this method
+    # to blow up and cause the server to continually fail to start.
+    # This is called very early from evm_application and ensures we attempt to call MiqWorker#kill
+    # to remove the problematic rows, and their in flight queue messages.
+    bad_workers = miq_workers.where.not(:type => MiqWorker.descendants.map(&:name))
+    if bad_workers.size.positive?
```

Comment:
I think this is clearer if we use `count` which forces a count query to the database.

```suggestion
    if bad_workers.count.positive?
```

Even so, I'm concerned reusing the query may cause issues if it caches things like the inheritance column in-between.  We might want a spec on this one if we don't already have one here.

Reply:
Oh I didn't realize this was old code in worker_management/monitor/kill.rb - carry on :)